### PR TITLE
Update MP validator to have same start and end date, only owner can edit

### PR DESCRIPTION
**Pull request by RohitKandimalla**
## MADiE PR

Jira Ticket: [MAT-4314](https://jira.cms.gov/browse/MAT-4314)
(Optional) Related Tickets:

### Summary

1. Disabling the Measurement Period Date field, if the current user is not the owner.
2. Validator is updated to accept MP start and end date to be the same.

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [x] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [x] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/models/MeasurementPeriodValidator.ts [hunk 1]
```diff
@@ -34,12 +34,10 @@ export const MeasurementPeriodValidator = Yup.object().shape({
     .when("measurementPeriodStart", (measurementPeriodStart, schema) => {
       if (measurementPeriodStart !== null) {
         if (!isNaN(measurementPeriodStart.getTime())) {
-          const dayAfter = new Date(
-            measurementPeriodStart.getTime() + 86400000
-          );
+          const dayAfter = new Date(measurementPeriodStart.getTime());
```
Comment (adongare) [inline src/models/MeasurementPeriodValidator.ts]: I think this is not `dayAfter` anymore. Could be `day` or just place this inline in `schema.min`?

Comment (RohitKandimalla) [reply src/models/MeasurementPeriodValidator.ts]: Yes you are right, I updated it to be an inline statement for schema

**Review by adongare — APPROVED** [review]
